src/data/user-monitoring/permission-fixer/PermissionFixerUserGroupD2Repository.ts

```diff
@@ -23,21 +24,45 @@ export class PermissionFixerUserGroupD2Repository implements PermissionFixerUser
             throw new UserGroupNotFoundException("Error getting user group: " + groupsIds);
         }
     }
-    async save(userGroup: PermissionFixerUserGroupExtended): Async<string> {
+    async save(userGroup: PermissionFixerUserGroupExtended, users: Ref[]): Async<string> {
         try {
-            const response = await this.api.models.userGroups.put(userGroup).getData();
-            if (response.status == "OK") {
+            const response = await this.appendUsersInUsergroup(userGroup, users);
+            if (response == "OK") {
                 log.info("Users added to minimal group");
             } else {
                 log.error("Error adding users to minimal group");
             }
 
-            log.info(JSON.stringify(response.response));
+            log.info(JSON.stringify(response));
 
-            return response.status;
+            return response;
         } catch (error) {
             console.debug(error);
             return "ERROR";
         }
     }
+
+    private async appendUsersInUsergroup(
+        userGroup: PermissionFixerUserGroupExtended,
+        users: Ref[]
+    ): Async<string> {
+        const usersIds = users.map(({ id }) => ({ id: id }));
+        const response = await this.api
+            .post<UserGroupResponse>(
+                `/userGroups/${userGroup.id}/users/`,
+                { async: false },
```

Comment:
Just a comment: as far as i know this does nothing as there is no async option for this endpoint.

Reply:
the reason to add tye async false is due without it I got a typo error

Reply:
I see, probably was developed with `api/metadata` in mind